### PR TITLE
Enable memcached by default for tests. Upgrade python-memcached

### LINKED ISSUE #8602: Use memcached in our unittests by default
We are using LocMemCache in our unittests, potentially for performance reasons. The problem is that we won't run into any of memcached problems, e.g a max cache-key length of 250 characters and others. (e.g https://github.com/mozilla/addons-server/issues/8598)

Let's use memcached by default.

### PR DESCRIPTION
**Pull request by EnTeQuAk**
* Adds a test to test unicode support for latest python-memcached
  versions

Fixes #8602

Let's see what travis says and which tests still need to be fixed. Shouldn't be too many.

### CODE REVIEW

File: src/olympia/lib/tests/test_cache.py [hunk 1]
```diff
@@ -65,6 +65,15 @@ def add(*args):
     assert add(1, 2) == cache.get(cache_key)
 
 
+def test_memcached_unicode():
+    """Regression test for
+
+    https://github.com/linsomniac/python-memcached/issues/79
+    """
+    cache.set('key', u'Iñtërnâtiônàlizætiøn2')
```
Comment (diox) [inline src/olympia/lib/tests/test_cache.py]: How about with an unicode key ?

Comment (EnTeQuAk) [reply src/olympia/lib/tests/test_cache.py]: Sure can do :)

The test was already failing on 1.57 so I was happy already.

**Comment by EnTeQuAk** [comment]
Meh, okay, I need to look into this more in detail. All these tests are passing just fine locally so something is up.

This looks like it's related to pytest-xdist so I'll try to come up with something that randomizes the memcached location based on xdist similarly to what pytest-django does.

**Comment by EnTeQuAk** [comment]
This is turning into a bigger refactor that I'll continue throughout the week. I'm closing this PR for now.

**Pull request closed without merging**